### ISSUE TITLE
Directly accessing exports of JS snippets

### ISSUE THREAD

**Issue by da-x** [rustwasm/wasm-bindgen#2820]
Suppose I have the following:

```rust
#[wasm_bindgen(module = "/js/some-rollup.js")]
extern "C" {
// ... various stuff here ...
}
```

And `some-rolleup.js` is a JS rollup that ends in the following export: 

```javascript
// ... various JS code here ...
export { SomeType, namespace };
```

Is there a way to directly obtain a `JsValue` that represents `SomeType` or `namespace`, besides adding dedicated accessory functions in the JS code that return them in addition to the export?

**Comment by AmateurECE** [comment]
I'm curious as well, is this an existing feature? It would go a long way towards facilitating calling JS from Rust in the browser in a more object-oriented fashion.

**Comment by Liamolucko** [comment]
Yes; it doesn't seem to be very well documented right now, but you can do this via `static`s:

```rust
#[wasm_bindgen(module = "/js/some-rollup.js")]
extern "C" {
    #[wasm_bindgen(js_name = SomeType)]
    static SOME_TYPE: JsValue;
}

#[wasm_bindgen]
pub fn retrieve() -> JsValue {
    // `SOME_TYPE` is expanded by the macro to a `JsStatic<JsValue>` (documented at https://docs.rs/wasm-bindgen/latest/wasm_bindgen/struct.JsStatic.html),
    // which implements `Deref<Target = JsValue>`.
    SOME_TYPE.clone()
}
```

**Comment by AmateurECE** [comment]
Thanks very much for your response! I was able to figure this out. Do you think users could benefit from adding something about this to the guide? If so, where in the guide should it live?

**Comment by Liamolucko** [comment]
> Do you think users could benefit from adding something about this to the guide? If so, where in the guide should it live?

Yeah, a page in the "Reference" section would make sense.

**Comment by daxpedda** [comment]
Fixed in #3127.
